lib/guard/puma/runner.rb

```diff
@@ -40,7 +40,8 @@ def initialize(options)
     end
 
     def start
-      if Gem.win_platform?
+      if ENV['SHELL'].nil? && !ENV['COMSPEC'].nil?
```

Comment:
Very nice. My last bit of feedback is to extract that to a private method, then I'll merge. Give it a good name 😉